catroid/src/createatschool/java/org/catrobat/catroid/createatschool/transfers/FetchTemplatesTask.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2016 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.createatschool.transfers;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.util.Log;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.dialogs.CustomAlertDialogBuilder;
+import org.catrobat.catroid.utils.Utils;
+import org.catrobat.catroid.web.ServerCalls;
+import org.catrobat.catroid.web.WebconnectionException;
+
+public class FetchTemplatesTask extends AsyncTask<Void, Void, String> {
+
+	private static final String TAG = FetchTemplatesTask.class.getSimpleName();
+
+	private Context context;
+	private ProgressDialog progressDialog;
+	private OnFetchTemplatesCompleteListener onFetchTemplatesCompleteListener;
+	private WebconnectionException exception;
+	private String message;
+
+	public FetchTemplatesTask(Context activity) {
+		this.context = activity;
+	}
+
+	public void setOnFetchTemplatesCompleteListener(OnFetchTemplatesCompleteListener listener) {
+		onFetchTemplatesCompleteListener = listener;
+	}
+
+	@Override
+	protected void onPreExecute() {
+		super.onPreExecute();
+		if (context == null) {
+			return;
+		}
+		String title = context.getString(R.string.please_wait);
+		String message = context.getString(R.string.fetching_templates);
+		progressDialog = ProgressDialog.show(context, title, message);
+	}
+
+	@Override
+	protected String doInBackground(Void... arg0) {
+		try {
+			if (!Utils.isNetworkAvailable(context)) {
+				exception = new WebconnectionException(WebconnectionException.ERROR_NETWORK, "Network not available!");
+				return null;
```

Comment:
Your suggestion is simply not possible. 
First, the doInBackground function overrides from AsyncTask. So you can't simply add a throws someException to the method signature. This doesn't even compile.
Second, the doInBackground function runs on a background thread. This means the error dialog can't be shown at that stage (it's not the UI thread). This can be done onPostExecute(..). The exception is stored as member variable to get the error message. 

Actually, this is the same implementation as in most other Tasks in the org.catrobat.catroid.transfers package.